controllers/ibu_controller.go

```diff
@@ -374,6 +374,13 @@ func validateStageTransition(ibu *lcav1alpha1.ImageBasedUpgrade, isAfterPivot bo
 
 func (r *ImageBasedUpgradeReconciler) updateStatus(ctx context.Context, ibu *lcav1alpha1.ImageBasedUpgrade) error {
 	ibu.Status.ObservedGeneration = ibu.ObjectMeta.Generation
+
+	// TODO: Does a blanket update of all conditions even make sense, or should
+	// we be more selective?
+	for i := range ibu.Status.Conditions {
+		ibu.Status.Conditions[i].ObservedGeneration = ibu.ObjectMeta.Generation
```

Comment:
We seem to be setting the `ObservedGeneration` properly when setting the status conditions via [this](https://github.com/openshift-kni/lifecycle-agent/blob/f3d164e4d31fe63e07c2214cc90efd6d1852c6c3/controllers/utils/conditions.go#L104). 

Perhaps you observed that `kubeclt/oc wait` for a specific status condition didn't work as expected because of an incorrect `ObservedGeneration` (my best guess is with the `Idle` condition)?

Reply:
Hmm, I guess it can happen whenever there is a spec change that doesn't trigger any condition updates. In theory, we probably should always update the observedGeneration field for the conditions related to the current desired stage.

Reply:
>my best guess is with the Idle condition)?

Yes my issue was that I was Idle aand the generation didn't match so wait for idle condition never ends

Reply:
@jc-rh I don't think there is any such changes, currently.

Reply:
> @jc-rh I don't think there is any such changes, currently.

an OADP or extra manifest update while in prep?